cmd/swapd/contract.go

```diff
@@ -79,37 +84,38 @@ func deploySwapFactory(
 
 	if (forwarderAddress == ethcommon.Address{}) {
 		// deploy forwarder contract as well
-		address, err := contracts.DeployGSNForwarderWithKey(ctx, ec, privkey)
+		var err error
+		forwarderAddress, err = contracts.DeployGSNForwarderWithKey(ctx, ec, privkey)
 		if err != nil {
 			return ethcommon.Address{}, nil, err
 		}
-		forwarderAddress = address
 	}
 
-	address, sf, err := contracts.DeploySwapFactoryWithKey(ctx, ec, privkey, forwarderAddress)
+	swapFactoryAddress, sf, err := contracts.DeploySwapFactoryWithKey(ctx, ec, privkey, forwarderAddress)
 	if err != nil {
 		return ethcommon.Address{}, nil, err
 	}
 
 	// store the contract address on disk
-	fp := path.Join(dataDir, "contract-address.json")
-	if err = writeContractAddressToFile(fp, address.String()); err != nil {
+	err = writeContractAddressToFile(
+		path.Join(dataDir, "contract-addresses.json"),
+		&contractAddresses{
+			SwapFactory: swapFactoryAddress,
+			Forwarder:   forwarderAddress,
+		},
+	)
+	if err != nil {
 		return ethcommon.Address{}, nil, fmt.Errorf("failed to write contract address to file: %w", err)
 	}
 
-	return address, sf, nil
+	return swapFactoryAddress, sf, nil
 }
 
 // writeContractAddressToFile writes the contract address to the given file
-func writeContractAddressToFile(fp, addr string) error {
-	bz, err := json.MarshalIndent(&struct {
-		ContractAddress string
-	}{
-		ContractAddress: addr,
-	}, "", "\t")
+func writeContractAddressToFile(filePath string, addresses *contractAddresses) error {
+	jsonData, err := json.MarshalIndent(addresses, "", "\t")
```

Comment:
It doesn't really matter, but I think I like 2 space indenting better than tabs for JSON after looking at it.